File: src/main/java/com/gmail/borlandlp/minigamesdtools/arena/gui/providers/GUIListener.java
package com.gmail.borlandlp.minigamesdtools.arena.gui.providers;

import com.gmail.borlandlp.minigamesdtools.Debug;
import com.gmail.borlandlp.minigamesdtools.arena.ArenaEventListener;
import com.gmail.borlandlp.minigamesdtools.arena.localevent.*;

public class GUIListener implements ArenaEventListener {
    private GUIController guiController;

    public GUIListener(GUIController guiController) {
        this.guiController = guiController;
    }

    @ArenaEventHandler
    public void onPlayerDeath(ArenaPlayerDeathLocalEvent event) {
        Debug.print(Debug.LEVEL.NOTICE, this.getClass().getSimpleName() + "#" + event);
    }

    @ArenaEventHandler
    public void onPlayerKilled(ArenaPlayerKilledLocalEvent event) {
        Debug.print(Debug.LEVEL.NOTICE, this.getClass().getSimpleName() + "#" + event);
    }

    @ArenaEventHandler
    public void onPlayerLeave(ArenaPlayerLeaveLocalEvent event) {
        Debug.print(Debug.LEVEL.NOTICE, this.getClass().getSimpleName() + "#" + event);
    }

    @ArenaEventHandler
    public void onPlayerJoin(ArenaPlayerJoinLocalEvent event) {
        Debug.print(Debug.LEVEL.NOTICE, this.getClass().getSimpleName() + "#" + event);
    }
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/arena/gui/providers/GUIListener.kt
package com.gmail.borlandlp.minigamesdtools.arena.gui.providers

import com.gmail.borlandlp.minigamesdtools.Debug
import com.gmail.borlandlp.minigamesdtools.Debug.print
import com.gmail.borlandlp.minigamesdtools.arena.ArenaEventListener
import com.gmail.borlandlp.minigamesdtools.arena.localevent.*

class GUIListener(private val guiController: GUIController) : ArenaEventListener {
    @ArenaEventHandler
    fun onPlayerDeath(event: ArenaPlayerDeathLocalEvent) {
        print(
            Debug.LEVEL.NOTICE,
            this.javaClass.simpleName + "#" + event
        )
    }

    @ArenaEventHandler
    fun onPlayerKilled(event: ArenaPlayerKilledLocalEvent) {
        print(
            Debug.LEVEL.NOTICE,
            this.javaClass.simpleName + "#" + event
        )
    }

    @ArenaEventHandler
    fun onPlayerLeave(event: ArenaPlayerLeaveLocalEvent) {
        print(
            Debug.LEVEL.NOTICE,
            this.javaClass.simpleName + "#" + event
        )
    }

    @ArenaEventHandler
    fun onPlayerJoin(event: ArenaPlayerJoinLocalEvent) {
        print(
            Debug.LEVEL.NOTICE,
            this.javaClass.simpleName + "#" + event
        )
    }

}